cyrw/ioctl.go

```diff
@@ -349,36 +348,11 @@ func (d *Device) pollForIoctl(buf []uint32) ([]byte, error) {
 	return nil, errors.New("pollForIoctl timeout")
 }
 
-// waitForCredit waits for a credit to use for the next transaction
-func (d *Device) waitForCredit(buf []uint32) error {
-	if d.has_credit() {
-		return nil
-	}
-	for retries := 0; retries < 10; retries++ {
-		avail, length := d.f2PacketAvail()
-		if !avail {
-			time.Sleep(10 * time.Millisecond)
-			continue
-		}
-		err := d.wlan_read(buf[:], int(length))
-		if err != nil {
-			return err
-		}
-		buf8 := u32AsU8(buf[:])
-		_, _, _, err = d.rx(buf8[:length])
-		if err != nil {
-			return err
-		}
-		return nil
-	}
-	return errors.New("waitForCredit timeout")
-}
-
 // check_status handles F2 events while status register is set.
 func (d *Device) check_status(buf []uint32) error {
 	d.trace("check_status")
 	for {
-		status := d.spi.Status()
+		status := d.status()
```

Comment:
status method does active update if stale, spibus's Status method does no IO if stale.

Reply:
It will not be stale because of a previous getInterrupt call or another cmd_read|write that cached the last value.

Also, this change deviates from the ref: now we're potentially going to read the STATUS_REGISTER unnecessarily.  The ref driver doesn't do that.

d.status() should probably go away, along with d.lastStatusGet.

d.spi.Status() is sufficient.

Reply:
Ah, the for loop confused me since I thought it was iterating on status check failed, but it's the other way around. changing back